pmd-dist/src/main/resources/scripts/pmd.bat

```diff
@@ -1,6 +1,54 @@
 @echo off
 set TOPDIR="%~dp0.."
 set OPTS=
-set MAIN_CLASS=net.sourceforge.pmd.PMD
+set COMMAND=%1
+set MAIN_CLASS=net.sourceforge.pmd.cli.PmdCli
 
-java %PMD_JAVA_OPTS% -classpath %TOPDIR%\conf;%TOPDIR%\lib\* %OPTS% %MAIN_CLASS% %*
+
+:: sets the jver variable to the java version, eg 901 for 9.0.1+x or 180 for 1.8.0_171-b11
+:: sets the jvendor variable to either java (oracle) or openjdk
+for /f tokens^=1^,3^,4^,5^ delims^=.-_+^"^  %%j in ('java -version 2^>^&1 ^| find "version"') do (
+  set jvendor=%%j
+  if %%l EQU ea (
+    set /A "jver=%%k00"
+  ) else (
+    set /A jver=%%k%%l%%m
+  )
+)
+
+Set "jreopts="
+:: oracle java 9 and 10 has javafx included as a module
+if /I "%jvendor%" EQU "java" (
+    if %jver% GEQ 900 (
+        if %jver% LSS 1100 (
+            :: enable reflection
+            Set jreopts=--add-opens javafx.controls/javafx.scene.control.skin=ALL-UNNAMED
+        )
+    )
+)
+
+set "_needjfxlib=0"
+if %COMMAND% EQU "designer" (
```

Comment:
I don't have a Windows machine, so I'm unable to really test this… I tried to unify all commands under this one (following the old designer one). Can anyone help me here?